packages/uix-guest/src/index.ts

```diff
@@ -128,7 +128,7 @@ export async function register<Outgoing extends GuestApis>(
   config: GuestConfigWithMethods<Outgoing>
 ) {
   const guest = new GuestServer(config);
-  await guest.register(config.methods, config.metadata);
+  guest.register(config.methods, config.metadata);
```

Comment:
in this case, we should also remove the sync word on the function declaration

Reply:
We can't due to backward compatibility reasons.

Reply:
I completely agree with @VladimirZaets

When a function is declared with the `async` keyword, it automatically returns a `Promise`, regardless of whether the function explicitly returns one.
This means that clients calling this function can rely on its return value being a Promise. They can chain `.then()` and `.catch()` methods to handle the resolved value or any errors.

If you remove the `async` keyword from the function declaration, the function no longer returns a Promise. Instead, it returns the `guest` object directly.

If clients were previously calling the `register` function expecting a Promise, they would likely have code like this:
```
register(config).then(guest => {
  // Do something with the guest object
}).catch(error => {
  // Handle any errors
});
```

^^^
If you remove `async`, the function no longer returns a Promise, and these `.then()` and `.catch()` methods would no longer work. **The client code would break because it's trying to call .then() on a non-Promise value.**

Reply:
@VladimirZaets ,
I have another question though: with this change the `register` function does not wait for `guest.register()` to complete before returning the `guest` object.
This could potentially lead to issues if the calling code assumes that registration **has been _fully_ completed** by the time the `guest` object is returned.
This can lead to race conditions or issues if the registration needs to be completed before proceeding. Are we sure we are safe here?

Reply:
I should have explained this better in the comment. using async in the function while not using await has no effect. In this case, the code is working because the function calling register() is an async function that uses await ex: (await register). To resume, is possible to use async without await but is not recommended, it leads to confusing, because it make you think the function is synchronous but in reality is not. Anyway, I guess it's the only way to solve the race condition for this case.